crates/threshold-signature-server/src/helpers/app_state.rs

```diff
@@ -0,0 +1,247 @@
+use crate::{
+    chain_api::{get_api, get_rpc, EntropyConfig},
+    launch::Configuration,
+    signing_client::ListenerState,
+    user::api::*,
+};
+use anyhow::anyhow;
+use entropy_kvdb::kv_manager::KvManager;
+use entropy_shared::X25519PublicKey;
+use sp_core::{crypto::AccountId32, sr25519, Pair};
+use std::{
+    collections::HashMap,
+    sync::{Arc, PoisonError, RwLock},
+};
+use subxt::{
+    backend::legacy::LegacyRpcMethods, tx::PairSigner, utils::AccountId32 as SubxtAccountId32,
+    OnlineClient,
+};
+use x25519_dalek::StaticSecret;
+
+/// Represents the state relating to the prerequisite checks
+#[derive(Clone, PartialEq, Eq)]
+pub enum TssState {
+    /// Initial state where no connection to chain node has been made
+    NoChainConnection,
+    /// Connection is made to the chain node but the account may not be yet funded
+    ReadOnlyChainConnection,
+    /// Fully ready and able to participate in the protocols
+    Ready,
+}
+
+impl TssState {
+    fn new() -> Self {
+        TssState::NoChainConnection
+    }
+
+    fn is_ready(&self) -> bool {
+        self == &TssState::Ready
+    }
+
+    fn can_read_from_chain(&self) -> bool {
+        self != &TssState::NoChainConnection
+    }
+}
+
+#[derive(Clone)]
```

Comment:
```suggestion
/// In-memory store of application state
#[derive(Clone)]
```
or something like that